File: pinyougou-web-user/src/main/java/cn/itcast/core/controller/UserController.java
package cn.itcast.core.controller;

import cn.itcast.core.entity.Result;
import cn.itcast.core.pojo.item.Item;
import cn.itcast.core.pojo.user.User;
import cn.itcast.core.service.user.UserService;
import cn.itcast.core.utils.phone.PhoneFormatCheckUtils;
import com.alibaba.dubbo.config.annotation.Reference;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.annotation.Resource;
import java.util.List;

@RestController
@RequestMapping("/user")
public class UserController {

    @Reference
    private UserService userService;

    @RequestMapping("/sendCode.do")
    public Result sendCode(String phone) {
        try {
            if (!PhoneFormatCheckUtils.isPhoneLegal(phone)) {
                return new Result(false, "手机号码格式不正确");
            }
            userService.sendCode(phone);
            return new Result(true, "发送成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "发送失败");
        }
    }

    @RequestMapping("/add.do")
    public Result add(String smscode, @RequestBody User user) {
        try {
            userService.add(smscode, user);
            return new Result(true, "注册成功");
        } catch (RuntimeException e) {
            e.printStackTrace();
            return new Result(false, e.getMessage());
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "注册失败");
        }
    }

    @RequestMapping("/findOneByUserName.do")
    public User findOneByUserName() {
        String userId = SecurityContextHolder.getContext().getAuthentication().getName();
        User one = userService.findOne(userId);
        if (one != null) {
            return one;
        }
        return null;
    }

    @RequestMapping("/updateUserInfo.do")
    public Result updateUserInfo(@RequestBody User user) {
        try {
            String userId = SecurityContextHolder.getContext().getAuthentication().getName();
            user.setUsername(userId);
            userService.updateUser(user);
            return new Result(true, "注册成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    //查询当前登陆账户的收藏商品
    @RequestMapping("/findAllConcern.do")
    public List<Item> getAll(){
        String userId = SecurityContextHolder.getContext().getAuthentication().getName();
        return userService.findAllConcern(userId);
    }

}

File: pinyougou-web-user/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller   ,userService,uploadService){
	$scope.entity = {};
	//注册用户
	$scope.reg=function(){
		
		//比较两次输入的密码是否一致
		if($scope.password!=$scope.entity.password){
			alert("两次输入密码不一致，请重新输入");
			$scope.entity.password="";
			$scope.password="";
			return ;			
		}
		//新增
		userService.add($scope.entity,$scope.smscode).success(
			function(response){
				alert(response.message);
			}		
		);
	}
    
	//发送验证码
	$scope.sendCode=function(){
		if($scope.entity.phone==null || $scope.entity.phone==""){
			alert("请填写手机号码");
			return ;
		}
		
		userService.sendCode($scope.entity.phone  ).success(
			function(response){
				alert(response.message);
			}
		);		
	}

	$scope.user={}
	//查询当前用户的信息
	$scope.findOneByUserName=function () {
		alert("111")
		userService.findOneByUserName().success(
			function (response) {
				$scope.user = response;
				$scope.user.headPic = response.headPic;
            }
		)
    }

    $scope.url;

    $scope.uploadPhoto = function(){
        // 调用uploadService的方法完成文件的上传=
        uploadService.uploadFile().success(function(response){
            if(response.flag){
                alert(response.message);
                // 获得url
                $scope.url =  response.message;
                $scope.user.headPic=$scope.url;
            }else{
                alert(response.message);
            }
        });
    }

    $scope.updateUserInfo = function () {
		userService.updateUserInfo($scope.user).success(
			function (response) {
				alert(response.flag);
            }
		)
    }
	
});	
